New build system

mlunoe opened the pull request:
Adding new build system, providing a lot more freedom for developers and introducing coding style checks as well as tests!!

![Marathon frontend build](https://s3.amazonaws.com/f.cl.ly/items/262I2I0M1L283x1K3G2l/Image%202015-01-10%20at%2020%3A40%3A32.png)

With this build system both Backbone and React was updated to the newest versions.

mlunoe commented:
@pyronicide I don't know if you have time, but could you take a look at this? I'll gladly walk you through what I did.

ConnorDoyle commented:
Is it possible to break this up into smaller changes?  This seems like too much for anyone to reasonably review.

mlunoe commented:
I reverted a lot of style changes, so the git file diff view seem more comprehensive, but the build fails because of style errors :(
I removed the build hook for that, and will add that as a separate PR
